### PR TITLE
Support images names formatted as docker.io/image:tag

### PR DESCRIPTION
**Pull request by divolgin**
#### What this PR does / why we need it:

1. Images hosted in docker hub can be formatted in different ways:
```
busybox:1.32
library/busybox:1.32
docker.io/library/busybox:1.32
```
This PR adds support for `docker.io/busybox:1.32`, which is also a valid way to format the same image ref.

2. Removes duplicate entries from the `images` in the kustomization.yaml files
3. Fixes references similar to ECR's 111122222333.dkr.ecr.us-east-1.amazonaws.com/frontend:v1.0.1, which were being prefixed with `docker.io` (but caused no harm)

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
* Fixes an issue that caused images formatted as `docker.io/image:tag` to not be [rewritten](https://docs.replicated.com/vendor/operator-referencing-images) when [upgrading applications](https://docs.replicated.com/enterprise/updating-apps) in airgapped environments.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

**Review by sgalsaleh — DISMISSED** [review]

File: pkg/base/replicated_test.go [hunk 1]
```diff
@@ -946,14 +953,14 @@ metadata:
     helm.sh/chart: postgresql-10.13.8
     app.kubernetes.io/instance: postgresql
     app.kubernetes.io/managed-by: Helm
-  namespace: 
+  namespace: ${POD_NAMESPACE}
```
Comment (divolgin) [inline pkg/base/replicated_test.go:956]: This and `TestGetenv` allow running tests in okteto.  This change is unrelated to the fix in the PR.

File: pkg/k8sutil/kustomization.go [hunk 2]
```diff
@@ -38,12 +38,20 @@ func (s kustPatches) Less(i, j int) bool {
 }
 
 func WriteKustomizationToFile(kustomization kustomizetypes.Kustomization, file string) error {
-	// we remove newTags from here... because...
 	cleanedImages := []kustomizetypes.Image{}
+
+	// Remove tags and deduplicate image list.
+	// Tags are removed because we don't want kustomize to change tags, only image names
```
Comment (sgalsaleh) [inline pkg/k8sutil/kustomization.go:44]: should this be true for digests as well?

**Review by sgalsaleh — APPROVED** [review]

File: pkg/k8sutil/kustomization.go (hunk 2 above)
Comment (divolgin) [reply pkg/k8sutil/kustomization.go:44]: The comment refers to this line `image.NewTag = ""` (which isn't new), so I guess digests would be there too.

Comment (divolgin) [reply pkg/k8sutil/kustomization.go:44]: Actually, no.  Kustomize image has a separate field for `Digest`.

Comment (sgalsaleh) [reply pkg/k8sutil/kustomization.go:44]: yeah, you shouldn't be able to change a digest anyways now that i think about it.
